pkg/pillar/cmd/tpmmgr/tpmmgr.go

```diff
@@ -631,32 +632,61 @@ func printCapability() {
 	}
 }
 
-func printPCRs() {
+func getQuote(nonce []byte) ([]byte, []byte, []types.PCRValue, error) {
+	if !etpm.IsTpmEnabled() {
+		//No TPM, not an error, return empty values
+		return nil, nil, nil, nil
+	}
+
 	rw, err := tpm2.OpenTPM(etpm.TpmDevicePath)
 	if err != nil {
-		fmt.Printf("Error opening TPM device: %s", err)
-		return
+		return nil, nil, nil, err
 	}
 	defer rw.Close()
-	for i := 0; i < 23; i++ {
-		pcrVal, err := tpm2.ReadPCR(rw, i, tpm2.AlgSHA256)
+	pcrs := make([]types.PCRValue, 0)
+
+	var i uint8
+	for i = 0; i < 15; i++ {
```

Comment:
Why the hardcoded max 15? Is that the max in the TCG specification? Would make sense to define as a CONST in any case.

Reply:
addressed it, changed it (to 0-23 per TCG spec), and made it a constant.